### DISCUSSION TITLE
`this` undefined in unit test of a class method

### DISCUSSION THREAD

**Discussion by devjared1** [cypress-io/cypress#20056] [Questions and Help] [answered]
I'm new to Cypress and having an issue. I'm trying to test a method of a class that refers to a parameter of the class with `this.parameter` but the test is failing stating that this is undefined. I'm using `cy.spy(object, 'methodName')` to mock the method and call it. Any help is much appreciated!
The class:
![image](https://user-images.githubusercontent.com/71391311/152562292-5df02cf9-751b-4d2b-b471-6d7c7e64c840.png)
Test file:
![image](https://user-images.githubusercontent.com/71391311/152562368-20912004-9edd-492b-a2d4-7c340d4c8dc9.png)
Error:
![image](https://user-images.githubusercontent.com/71391311/152562457-48e33638-7ebd-41e3-88e3-ce99b610a5fe.png)

**Comment by bahmutov** [comment, marked as answer]
You are not preserving the `this` correctly when calling the method via function. See https://github.com/bahmutov/cypress-examples/commit/c0d239d6af3d3474a5f26d82e5582e153f02f55b and https://youtu.be/g_7LZXuEIMA
